Automattic/woocommerce-payments
# Add new fraud protection section to WCPay settings

Fixes Automattic/woocommerce-payments-server#2796 on the server repo.

#### Changes proposed in this Pull Request



Add new fraud protection section to WCPay settings


***Note: This section is currently behind a feature flag. See PR #5645 for info on how to enable the fraud and risk tools feature flag.***

This PR adds the standard and high fraud protection settings section user interface to the WCPay settings screen. This new section will allow merchants to select between the standard ruleset and the high ruleset. The new section also features a modal to provide information about each ruleset.





#### Testing instructions





* Check out this branch.
* All tests should pass.
* Login to your local client site and make sure your store's currency is set to USD.
* Visit the WCPay settings screen: `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
* Scroll to the bottom of the page. If you don't see the fraud protections section below the deposits section, then the feature flag is not enabled. See below for the command to run to enable it.

![No fraud protection settings section rendered](https://user-images.githubusercontent.com/60988591/219794624-3c11abee-4d24-4b4e-b48e-5b135f9679c8.png)

* Run the following command to enable the section to be rendered: `npm run wp option add wcpay_fraud_protection_settings_active 1`
* Refresh the settings page and scroll to the bottom. The fraud protection section should now be rendered below the deposits section.

![Fraud protection section rendered below deposits](https://user-images.githubusercontent.com/60988591/219795464-232c07bc-f735-4275-ab37-da5e8f208982.png)

* Evaluate the section: 
  * Check that there are no errors in the browser console. 
  * Check the text and layout for issues. 
  * Click the question mark icon beside each option to ensure that the correct modal displays. 
  * Ensure that modals close when clicking 'Got it' button, 'X' close button, and clicking outside of the modal.

* Navigate to the WooCommerce general settings page and change you currency to a currency other than USD. 
* Navigate back to the WCPay settings page and open each fraud protection modal. The rule referencing `$1,000.00` should now show a slightly modified text mentioning the equivalent of $1,000.00 in `$store_currency`.
 
* Check the section on mobile. Tunnel your local client site so you can reach it via the public internet.
* Visit your client site on your smartphone and login as the administrator.
* Navigate to the WCPay settings screen and scroll to the bottom of the page. 
* Evaluate the section on mobile: 
  * Check the text and layout for issues. 
  * Click the question mark icon beside each option to ensure that the correct modal displays. 
  * Ensure that modals close when clicking 'Got it' button and the 'X' close button.

Once you are done testing, if you would like to disable the feature flag, simply set the option to `0` or delete the option:
* `npm run wp option update wcpay_fraud_protection_settings_active 0`
* `npm run wp option delete wcpay_fraud_protection_settings_active`

***Note: The "Learn more about risk filtering" link also does not currently link anywhere as the documentation has yet to be published.***

-------------------


**Post merge**
